django/contrib/admin/templates/admin/change_form.html

```diff
@@ -24,19 +24,7 @@
 {% endif %}
 
 {% block content %}<div id="content-main">
-{% block object-tools %}
-{% if change %}{% if not is_popup %}
-  <ul class="object-tools">
-    {% block object-tools-items %}
-    <li>
-        {% url opts|admin_urlname:'history' original.pk|admin_urlquote as history_url %}
-        <a href="{% add_preserved_filters history_url %}" class="historylink">{% trans "History" %}</a>
-    </li>
-    {% if has_absolute_url %}<li><a href="{{ absolute_url }}" class="viewsitelink">{% trans "View on site" %}</a></li>{% endif %}
-    {% endblock %}
-  </ul>
-{% endif %}{% endif %}
-{% endblock %}
+{% block object-tools %}{% change_form_object_tools %}{% endblock %}
```

Comment:
Is there a need to add another template/template tag for cases liek this when you can already override the template block? I share Collin's concern that adding more levels of indirection makes things more complicated.

Reply:
Basically I've changed all object-tools with my version on root admin template, and for some model I've added a custom one (essentially an `{% extend "admin/change_{list,form}_object_tools.html" %}`.
I find simpler than copying in my project `/admin/change_{list,form}.html` (cannot extends from django `/admin/change_{list,form}.html` if I recall correctly), as I don't have to copy unnecessary template data. It's only an annoyance, not a real problem.
Can rid both commit if unnecessary.

Reply:
So you are proposing
```html+django
{% block object-tools %}{% if change %}{% if not popup %}
<ul class="object-tools">
{% block object-tools-items %}{% change_form_object_tools %}{% endblock %}
</ul>
{% endif %}{% endif %}{% endblock %}
```
It doesn't break compatibility, but it is only a minor override. I really don't mind about breaking change (admin is a tool, so I think I could live with that), but if you need more customization you can always override the change_X.html templates.
Tomorrow I'll rewrite this part.

Reply:
Rewritten, now backward compatibility is preserved.